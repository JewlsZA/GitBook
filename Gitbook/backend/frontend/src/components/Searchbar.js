import fetch from "isomorphic-fetch"
import React, { Component } from "react"
import { Button, Form } from "react-bootstrap"
// local imports
import "../css/loading.css"
import SearchResult from "./SearchResult"
import Loading from "./Loading"

class Searchbar extends Component {
  constructor() {
    super()
    this.state = {
      devUsername: "",
      usersInfo: [],
      infoLoaded: null,
      loadingstate: false,
    }
    this.onSubmitClick = this.onSubmitClick.bind(this)
  }

  // When on submit button is click, searched username is send to backend in data parameter
  // backend returns array of users information
  onSubmitClick = (e) => {
    this.setState({
      loadingstate: true,
    })
    e.preventDefault()
    fetch("/users", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        devUsername: this.state.devUsername,
      }),
    })
      .then((response) => response.json())
      .then((data) => {
        let myArr = data
        this.setState({
          usersInfo: myArr,
          infoLoaded: 1,
          loadingstate: false,
        })
      })
  }

  render() {
    return (
      <div style={{ padding: "30px" }}>
        <h1>Welcome to GitBook</h1>
        <br />
        {/* user input sets devUsername state */}
        <Form.Control
          value={this.state.devUsername}
          style={{ width: "80%", textAlign: "center", margin: "auto" }}
          type="input"
          placeholder="Search developers here by name"
          onChange={(e) =>
            this.setState({ devUsername: e.target.value.toLowerCase() })
          }
        />
        <br />
        {/* Submit Button */}
        {/* Triggers onClick*/}
        <Button onClick={this.onSubmitClick}>Search</Button>
        {/* Shows the loading component */}
        {this.state.loadingstate === true ? (
        <Loading />
        ) : null}
        {/* Renders the searched username information in a list */}
        {this.state.infoLoaded !== null ? (
          <SearchResult usersInfo={this.state.usersInfo}  />
        ) : null}
      </div>
    )
  }
}

export default Searchbar
