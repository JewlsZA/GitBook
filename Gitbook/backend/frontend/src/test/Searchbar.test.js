// import { render, screen } from '@testing-library/react';
import Searchbar from "../components/Searchbar"
import renderer from "react-test-renderer"
//import {onSubmitClick} from '../components/Searchbar'
import fetch from "isomorphic-fetch"
//import { expect } from 'chai';

// Extend timeout incase for slow internet connection, no failed tests
jest.setTimeout(60000)

test("renders correctly", () => {
  const tree = renderer.create(<Searchbar />).toJSON()
  expect(tree).toMatchSnapshot()
})

test("frontend fetches user information from the backend api", () => {
  fetch("http://localhost:8080/usertesting/jewlsza")
    .then((response) => response.json())
    .then((data) => {
      expect(data).toBeDefined()
    })
})
