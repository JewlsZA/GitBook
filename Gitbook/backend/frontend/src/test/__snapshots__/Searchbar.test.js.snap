// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "padding": "30px",
    }
  }
>
  <h1>
    Welcome to GitBook
  </h1>
  <br />
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Search developers here by name"
    style={
      Object {
        "margin": "auto",
        "textAlign": "center",
        "width": "80%",
      }
    }
    type="input"
    value=""
  />
  <br />
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
</div>
`;
